File: Selenium/Project/HRM_Activity4.py
import time

from selenium import webdriver
from selenium.webdriver.common.by import By
from selenium.webdriver.firefox.service import Service as FirefoxService
from selenium.webdriver.support import expected_conditions
from selenium.webdriver.support.wait import WebDriverWait
from webdriver_manager.firefox import GeckoDriverManager

service = FirefoxService(GeckoDriverManager().install())

with webdriver.Firefox(service=service) as driver:

    wait = WebDriverWait(driver, 10)
    driver.get("http://alchemy.hguy.co/orangehrm")
    driver.find_element(By.ID,"txtUsername").send_keys("orange")
    driver.find_element(By.ID,"txtPassword").send_keys("orangepassword123")
    driver.find_element(By.ID,"btnLogin").click()
    time.sleep(2)

    #click on PIM
    driver.find_element(By.XPATH,"//div/ul/li[2]/a/b").click()

    #click on Add Employee button
    time.sleep(2)
    driver.find_element(By.ID,"btnAdd").click()

    #Fill all the required fields
    fname=driver.find_element(By.NAME,"firstName")
    lname=driver.find_element(By.NAME,"lastName")
    fname.send_keys("Mounika")
    lname.send_keys("Amballa")
    driver.find_element(By.ID,"btnSave").click()

    #check name in the Employee List
    driver.find_element(By.ID,"menu_pim_viewEmployeeList").click()
    time.sleep(2)

    driver.find_element(By.ID,"empsearch_employee_name_empName").send_keys("Mounika Amballa")

    wait.until(expected_conditions.element_to_be_clickable((By.XPATH,"//div[@class='ac_results']//ul//li[1]//strong")))

    driver.find_element(By.XPATH,"//div[@class='ac_results']//ul//li[1]//strong").click()

    driver.find_element(By.XPATH,"//input[@id='searchBtn']").click()

    wait.until(expected_conditions.visibility_of_element_located((By.XPATH,"//tbody/tr[1]/td[3]")))

    # verify employee creation
    result=driver.find_element(By.XPATH,"//tbody/tr[1]/td[3]")

    if(result.text=="Mounika"):
        print("employee created")
    else:
        print("employee not created")

    driver.close()

File: Selenium/Project/HRM_Activity4.java
package Project;

import io.github.bonigarcia.wdm.WebDriverManager;
import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.firefox.FirefoxDriver;
import org.openqa.selenium.support.ui.ExpectedConditions;
import org.openqa.selenium.support.ui.WebDriverWait;

import java.time.Duration;


public class HRM_Activity4 {

    public static void main(String[] args) throws InterruptedException {
        WebDriverManager.firefoxdriver().setup();
        WebDriver driver = new FirefoxDriver();
        WebDriverWait wait=new WebDriverWait(driver, Duration.ofSeconds(5));
        driver.get("http://alchemy.hguy.co/orangehrm");
        driver.findElement(By.id("txtUsername")).sendKeys("orange");
        driver.findElement(By.id("txtPassword")).sendKeys("orangepassword123");
        driver.findElement(By.id("btnLogin")).click();
        Thread.sleep(2000);
        //click on PIM
        driver.findElement(By.xpath("//div/ul/li[2]/a/b")).click();

        //click on Add Employee button
        driver.findElement(By.id("btnAdd")).click();
        //Fill all the required fields
        WebElement fname=driver.findElement(By.name("firstName"));
        WebElement lname=driver.findElement(By.name("lastName"));
        fname.sendKeys("Mounika");
        lname.sendKeys("Amballa");
        driver.findElement(By.id("btnSave")).click();
        // check name in the Employee List
        driver.findElement(By.id("menu_pim_viewEmployeeList")).click();
        Thread.sleep(2000);
        driver.findElement(By.id("empsearch_employee_name_empName")).sendKeys("Mounika Amballa");

        driver.findElement(By.xpath("//div[@class='ac_results']//ul//li[1]//strong")).click();


        wait.until(ExpectedConditions.elementToBeClickable(By.xpath("//input[@id='searchBtn']")));
        driver.findElement(By.xpath("//input[@id='searchBtn']")).click();

        wait.until(ExpectedConditions.visibilityOfElementLocated(By.xpath("//tbody/tr[1]/td[3]")));

        // verify employee creation
        WebElement result=driver.findElement(By.xpath("//tbody/tr[1]/td[3]"));
        if(result.getText().contains("Mounika"))
        {
            System.out.println("Employee created");
        }
        else {
            System.out.println("Employee not created");

        }

        driver.close();

    }

}
